src/intrinsics/fb-www/fb-mocks.js

```diff
@@ -73,18 +68,12 @@ const fbMagicGlobalObjects = ["JSResource", "Bootloader"];
 
 function createMagicGlobalFunction(realm: Realm, global: ObjectValue | AbstractObjectValue, functionName: string) {
   global.$DefineOwnProperty(functionName, {
-    value: new NativeFunctionValue(realm, functionName, functionName, 0, (context, [string]) => {
-      invariant(string instanceof StringValue);
-      let value = `${functionName}("${string.value}")`;
-      let functionValue;
-
-      if (realm.fbLibraries.other.has(value)) {
-        functionValue = realm.fbLibraries.other.get(value);
-      } else {
-        functionValue = createAbstract(realm, "function", value);
-        realm.fbLibraries.other.set(value, functionValue);
-      }
-      invariant(functionValue instanceof AbstractValue);
+    value: new NativeFunctionValue(realm, functionName, functionName, 0, (context, args) => {
+      let functionValue = createAbstract(realm, "function");
+      functionValue.args = args;
+      // build the magic global function up manually from the arguments we pass it
+      // i.e. cx(...arg)
+      functionValue._buildNode = _args => t.callExpression(t.identifier(functionName), ((_args: any): Array<any>));
```

Comment:
This looks like a private API. Would be nice to verify it won't break with someone. `: any` probably isn't great either.
Maybe there's a supported way to do this? @NTillmann

Reply:
There is a way via creating the `AbstractValue` via the constructor route, but that means more boilerplate code than this route. This is already a somewhat hacky route as what we're doing isn't really "normal" – in that we want to serialize something in a very strict manner.